tests/datasource/fluent/test_sqlite_datasource.py

```diff
@@ -119,6 +119,8 @@ def create_sqlite_source() -> (
     return _create_sqlite_source
 
 
+# TODO: figure this out before merging
+@pytest.mark.xfail(reason="why are these tests are now failing???")
```

Comment:
The only thing I can think is that SQLite dbs can be in-memory and maybe it was tossed out somehow? I would think from your changes it would be the opposite - we are not preserving in-memory dbs by not closing the engine connection.

Reply:
Maybe SQL Alchemy is not installed for the sqlite tests (it's include in base python). If not, then it looks like `_poolclass = None`.

Reply:
I'm not sure exactly what was happening.
But I've reverted my changes, and made it so that SQLite retains the old behavior of creating 2 `Engine` objects. 
1 for testing the connection
1 for everything else.

That somehow fixed it.